doc/mloginfo.rst

```diff
@@ -393,11 +396,106 @@ For example:
    invoice-prod.invoices     insert       12768411     22233323        86313                12344
    invoice-prod.invoices     insert       12868411     22233323        86313                12344
 
+Sharding (``--sharding``)
+-----------------------------------------
+
+Outputs information of sharding configuration. Includes groupings of
+similar error/warning messages that are specific to sharding. A breakdown
+of chunk migration activity is included, specifying the source/destination
+shard, the number of chunk migrations attempted within the hour, and the
+ratio of successful to failed migrations. Failed chunk migrations specify
+when the failure occurred in formatted as [hours:minutes:seconds:milliseconds]
+but also notes if it precedes to be successful in the future (within the hour
+time frame). It also outputs chunk split statistics, which has a similar output
+as the chunk migration tables.
+
+For example:
+
+.. code-block:: bash
+
+   mloginfo mongod.log --sharding
+
+In addition to the default information, this command also outputs sharding
+related configurations, outputs a list of error/warning messages grouped by
+similar patterns, chunk migrations which have sent and received by the
+associated node in the log and chunk split statistics grouping by the hour.
+This can all be found below.
+
+.. code-block:: bash
+
+   SHARDING
+
+   Overview:
+
+      The role of this node: (shard)
+      Shards:
+         shard01: example:27018,example:27019,example:27020
+         shard02: example:27021,example:27022,example:27023
+         shard03: example:27024,example:27025,example:27026
+      CSRS:
+         configRepl: example:27033
+
+``--errors``
+^^^^^^^^^^^^
+
+This option toggles if sharding related errors/warnings are
```

Comment:
"This option toggles" sounds like the option will be triggered automatically. It would be clearly to describe the action, eg: "This option can be added to `--sharding` to include a summary of any sharding-related errors or warnings".

Since errors are normally hard fails and warnings are not ... this may want to be renamed or split into several options such as `--errors`, `--warnings`, and `--sharding-verbose` (which could imply all sharding details).

Reply:
I re-worded to indicate its an addition of information when added with `--sharding`